### PR TITLE
Replace many instances of sanity, sane, and insane

### PR DESCRIPTION
**Pull request by twostraws**
This replaces many instances of the words "sanity", "sane", and "insane" with equivalents that are less likely to hurt others. For example, "sanity check" has become "safety check", and "sane" has become "sensible". I hope this change follows the spirit of other similar initiatives, not least Apple's updates to coding terminology at https://developer.apple.com/news/?id=1o9zxsxl

I have done my best to restrict these changes to comments and documentation, which means there are still some instances of "sanity", "sane", and "insane" in places where I judged it would affect functionality somehow. Replacing these might also be straightforward, but I figured that's a separate change.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by xwu** [comment]
I thought we had standardized on “internal invariant” to replace “sanity check”?

**Comment by twostraws** [comment]
@xwu The original text described someone else's work as "insane", so I was hoping that "ridiculous" was an improvement. Perhaps the best change here would be for someone to amend the comment to describe the actual underlying problem this comment is describing.

**Comment by DougGregor** [comment]
> I thought we had standardized on “internal invariant” to replace “sanity check”? Safety is a loaded term in Swift.

I hate to say it because it's a lot of places to update a second time, but I agree. "Internal invariant" is the better replacement for "sanity" when we're talking about checking, because we carefully sort "internal invariants that we know hold but nonetheless check in debug builds to help ourselves" vs "checks we need to keep maintain the safety the language (e.g., integer overflow checking)". The use of the term for the former is ambiguous.

File: lib/Sema/DebuggerTestingTransform.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ class DiscriminatorFinder : public ASTWalker {
   }
 };
 
-/// Instrument decls with sanity-checks which the debugger can evaluate.
+/// Instrument decls with safety-checks which the debugger can evaluate.
```
Comment (DougGregor) [inline lib/Sema/DebuggerTestingTransform.cpp:66]: Here's a concrete example with "safety" being ambiguous. We're actually talking about checking invariants, but in a debugger context "safety" could absolutely mean the safety checking Swift does by default.

File: stdlib/public/core/Assert.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 /// Performs a traditional C-style assert with an optional message.
 ///
-/// Use this function for internal sanity checks that are active during testing
+/// Use this function for internal safety checks that are active during testing
```
Comment (DougGregor) [inline stdlib/public/core/Assert.swift:15]: Another instance: this isn't about safety, it's about invariants.

File: stdlib/public/core/Builtin.swift [hunk 3]
```diff
@@ -678,7 +678,7 @@ internal func _isUnique<T>(_ object: inout T) -> Bool {
 }
 
 /// Returns `true` if `object` is uniquely referenced.
-/// This provides sanity checks on top of the Builtin.
+/// This provides safety checks on top of the Builtin.
```
Comment (DougGregor) [inline stdlib/public/core/Builtin.swift:681]: Same comment here.

File: stdlib/public/core/ContiguousArrayBuffer.swift [hunk 4]
```diff
@@ -670,7 +670,7 @@ internal struct _ContiguousArrayBuffer<Element>: _ArrayBufferProtocol {
 
   /// Returns `true` iff this buffer's storage is uniquely-referenced.
   ///
-  /// This function should only be used for internal sanity checks.
+  /// This function should only be used for internal safety checks.
```
Comment (DougGregor) [inline stdlib/public/core/ContiguousArrayBuffer.swift:673]: Same here.

File: stdlib/public/core/SliceBuffer.swift [hunk 5]
```diff
@@ -298,7 +298,7 @@ internal struct _SliceBuffer<Element>
 
   /// Returns `true` iff this buffer's storage is uniquely-referenced.
   ///
-  /// This function should only be used for internal sanity checks and for
+  /// This function should only be used for internal safety checks and for
```
Comment (DougGregor) [inline stdlib/public/core/SliceBuffer.swift:301]: Same here.

**Review by DougGregor — COMMENTED** [review]
Thank you for working on this! I left a couple of comments on places where "safety" becomes ambiguous, but didn't get to every such change (sorry).

**Comment by twostraws** [comment]
I would be happy to convert all "sanity check -> "safety check" changes over to "sanity check" -> "internal invariant check". Am I okay to proceed with that, and push more changes back to my branch?

**Comment by twostraws** [comment]
I am inclined to close this PR; it seems to be languishing.

**Comment by rjmccall** [comment]
If we can resolve the conflicts, we can get this in.

**Comment by twostraws** [comment]
@rjmccall Thank you! I have resolved the conflict, taken into account the feedback from both @xwu and @DougGregor, and pushed more changes. However, there may still be places where "safety check" remains and is ambiguous – I am happy to address those if identified, or of course someone else can do the same.

File: validation-test/stdlib/Unicode.swift.gyb [hunk 6]
```diff
@@ -1689,7 +1689,7 @@ StringCookedViews.test("UTF8ForContiguousUTF16/StringsWithUnpairedSurrogates") {
 }
 
 func verifyThatStringIsOpaqueForCoreFoundation(_ nss: NSString) {
-  // Sanity checks to make sure we are testing the code path that does UTF-8
+  // Safety checks to make sure we are testing the code path that does UTF-8
```
Comment (xwu) [inline validation-test/stdlib/Unicode.swift.gyb:1692]:
```suggestion
  // Checks to make sure we are testing the code path that does UTF-8
```

File: validation-test/stdlib/Unicode.swift.gyb [hunk 7]
```diff
@@ -1770,7 +1770,7 @@ StringCookedViews.test("UTF8ForNonContiguousUTF16Extra") {
       kCFAllocatorDefault, bytes, bytes.count,
       CFStringBuiltInEncodings.macRoman.rawValue, false, kCFAllocatorNull)
 
-    // Sanity checks to make sure we are testing the code path that does UTF-8
+    // Safety checks to make sure we are testing the code path that does UTF-8
```
Comment (xwu) [inline validation-test/stdlib/Unicode.swift.gyb:1773]:
```suggestion
    // Checks to make sure we are testing the code path that does UTF-8
```

File: validation-test/stdlib/Unicode.swift.gyb [hunk 8]
```diff
@@ -1793,7 +1793,7 @@ StringCookedViews.test("UTF8ForNonContiguousUTF16Extra") {
     var cfstring: CFString = CFStringCreateWithBytes(kCFAllocatorDefault,
         bytes, bytes.count, CFStringBuiltInEncodings.macRoman.rawValue, false)
 
-    // Sanity checks to make sure we are testing the code path that does UTF-8
+    // Safety checks to make sure we are testing the code path that does UTF-8
```
Comment (xwu) [inline validation-test/stdlib/Unicode.swift.gyb:1796]:
```suggestion
    // Checks to make sure we are testing the code path that does UTF-8
```

**Review by xwu — COMMENTED** [review]
I feel very strongly that "safety check" is not helpful or, worse, misleading where it's used in this PR. In general, if there's an assertion, then it's an internal invariant check. Otherwise, in most places, it can be replaced with a descriptor like "basic" or "cursory," or oftentimes just omitted altogether.

There's one change, for example, that causes Swift to inform users that the standard library is being compiled without safety checks (implying lack of memory safety), when in fact what's disabled are internal invariant checks.

**Comment by twostraws** [comment]
@xwu I'm not in a position to say where each choice would be better, but I am grateful that you have taken the time to provide such comprehensive feedback to help improve the whole PR. Thank you!

**Comment by twostraws** [comment]
This seems to be going nowhere, so regretfully I'm closing it.

**Pull request closed without merging**